infrastructure/src/main/java/org/corfudb/infrastructure/log/InMemoryStreamLog.java

```diff
@@ -61,14 +63,11 @@ public synchronized void append(long address, LogData entry) {
             throwLogUnitExceptionsIfNecessary(address, entry);
         }
         logCache.put(address, entry);
-
-
-        globalTail.getAndUpdate(maxTail -> entry.getGlobalAddress() > maxTail
-                ? entry.getGlobalAddress() : maxTail);
+        logMetadata.update(entry);
     }
 
     private boolean isTrimmed(long address) {
-        if (address < startingAddress) {
+        if (address < startingAddress || trimmed.contains(address)) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)